pwndbg/dbg/lldb/repl/fuzzy.py

```diff
@@ -0,0 +1,307 @@
+# Licensed under the MIT License
+# Modified from https://github.com/lebr0nli/GEP
+# Copyright (c) 2022 Alan Li
+# Copyright (c) 2025 Zhi-Qiang Zhou
+
+import lldb
+
+import atexit, os, re, threading, sys, tempfile, shutil
+import functools
+
+from typing import Iterator, Callable
+from subprocess import PIPE, Popen
+
+from prompt_toolkit import ANSI
+from prompt_toolkit import PromptSession
+from prompt_toolkit.application import run_in_terminal
+from prompt_toolkit.completion import CompleteEvent
+from prompt_toolkit.completion import Completer
+from prompt_toolkit.completion import Completion
+from prompt_toolkit.document import Document
+from prompt_toolkit.history import FileHistory
+from prompt_toolkit.key_binding import KeyPressEvent
+from prompt_toolkit.key_binding import KeyBindings
+from prompt_toolkit.output import create_output
+
+from pwndbg.dbg.lldb import LLDB
+
+# global variables
+HAS_FZF = shutil.which("fzf") is not None
+PROMPT = ANSI("\x1b[34mpwndbg-lldb> ")
+HISTORY_FILE = os.path.expanduser("~/.pwndbg_history")
+
+FZF_RUN_CMD = (
+    "fzf",
+    "--select-1",
+    "--exit-0",
+    "--tiebreak=index",
+    "--no-multi",
+    "--height=40%",
+    "--layout=reverse",
+)
+
+FZF_PRVIEW_WINDOW_ARGS = (
+    "--preview-window",
+    "right:55%:wrap",
+)
+
+
+def get_lldb_completes(dbg: LLDB, query: str="") -> list[str]:
+    interp: lldb.SBCommandInterpreter = dbg.debugger.GetCommandInterpreter()
+    matches = lldb.SBStringList()
+    _num_matches = interp.HandleCompletion(query, len(query), 0, -1, matches)
+    proposals = [matches.GetStringAtIndex(i) for i in range(1, matches.GetSize())]
+    return [i for i in proposals]
+
+
+def safe_get_help_docs(dbg: LLDB, command: str) -> str | None:
+    interp: lldb.SBCommandInterpreter = dbg.debugger.GetCommandInterpreter()
+    add_to_history = False
+    result = lldb.SBCommandReturnObject()
+    interp.HandleCommand(f"help {command}", result, add_to_history)
+    return result.GetOutput()
```

Comment:
While this is better in that it doesn't call that function, I still would like to know if there really is no way to do this without invoking an LLDB command.

Reply:
As far as I know, the [CommandObjectHelp.cpp](https://github.com/llvm/llvm-project/blob/main/lldb/source/Commands/CommandObjectHelp.cpp) is used to generate help docs, however the class is not exposed.

Reply:
What about `HandleCompletionWithDescription`?

Reply:
The `HandleCompletionWithDescriptions` only generate a brief description:

<img width="564" alt="Screenshot 2025-06-05 at 05 44 02" src="https://github.com/user-attachments/assets/c9c0dd5d-a7b3-4981-8a2f-6e6ab2b2ed92" />

However, with help:

<img width="560" alt="Screenshot 2025-06-05 at 05 46 09" src="https://github.com/user-attachments/assets/ffa4e008-b7a7-4ec8-8f0f-579387dbfa53" />

Reply:
Alrighty, then. I think resorting to the command in this case is justified.